src/TConsole.cpp

```diff
@@ -814,13 +814,24 @@ void TConsole::closeEvent(QCloseEvent* event)
 
     if (!mUserAgreedToCloseConsole) {
     ASK:
-        int choice = QMessageBox::question(this, tr("Save profile?"), tr("Do you want to save the profile %1?").arg(mProfileName), QMessageBox::Yes | QMessageBox::No | QMessageBox::Cancel);
+        QMessageBox messageBox = QMessageBox(this);
+        messageBox.setIcon(QMessageBox::Question);
+        messageBox.setWindowTitle(tr("Save profile?"));
+        messageBox.setText(tr("Do you want to save the profile %1?").arg(mProfileName));
+        messageBox.addButton(QMessageBox::Yes);
+        QPushButton* alwaysButton = messageBox.addButton(tr("Yes, &always"), QMessageBox::YesRole);
```

Comment:
You are adding *another* button with the `YesRoll`, as such, it is not possible to determine which of them will become the ***default*** button. You need to capture the address of the button on the previous line so that you can specify it as the default one - otherwise this one might be used by accident - I suggest:
```diff
-        messageBox.addButton(QMessageBox::Yes);
-        QPushButton* alwaysButton = messageBox.addButton(tr("Yes, &always"), QMessageBox::YesRole);
+        auto yesButton = messageBox.addButton(QMessageBox::Yes);
+        auto alwaysButton = messageBox.addButton(tr("Yes, &always"), QMessageBox::YesRole);
+        messageBox.setDefault(yesButton);
```

Reply:
Outdated, as Vadi suggested restored original dialog with changing default value of an option.